File: frontend/src/api/gantt.ts
import { apiJson } from './client'

export type GanttTask = {
  id: string
  title: string
  status: string
  assignee_id: string | null
  start_at: string
  end_at: string
  blocked_by: string[]
}

export type GanttLink = {
  from_task_id: string
  to_task_id: string
  type: string
}

export type GanttData = {
  project_id: string
  project_name: string
  range_start: string
  range_end: string
  tasks: GanttTask[]
  links: GanttLink[]
}

export function fetchProjectGantt(token: string, projectId: string): Promise<GanttData> {
  return apiJson<GanttData>(`/projects/${encodeURIComponent(projectId)}/gantt`, token)
}

File: backend/app/schemas/gantt.py
"""Схемы данных для диаграммы Ганта."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class GanttTaskRead(BaseModel):
    id: UUID
    title: str
    status: str
    assignee_id: UUID | None
    start_at: datetime
    end_at: datetime
    blocked_by: list[UUID] = Field(default_factory=list)


class GanttLinkRead(BaseModel):
    from_task_id: UUID
    to_task_id: UUID
    type: str


class GanttRead(BaseModel):
    project_id: UUID
    project_name: str
    range_start: datetime
    range_end: datetime
    tasks: list[GanttTaskRead]
    links: list[GanttLinkRead]
